### PR TITLE
Table sorting

### PR DESCRIPTION
**Pull request by edaena**
Closes https://github.com/microsoft/bedrock/issues/1298

### CODE REVIEW

**Comment by samiyaakhtar** [comment]
Preview: 
sort-gif

**Comment by andrebriggs** [comment]
@edaena @samiyaakhtar Have does this behave with filters and a large number of rows? Can you provide  link to a deployed version?

File: src/client/DeploymentTable.tsx [hunk 1]
```diff
@@ -35,8 +40,38 @@ interface ITableProps {
...
+  ((item1: IDeploymentField, item2: IDeploymentField) => number) | null
+>;
+
+const sortingBehavior = new ColumnSorting<IDeploymentField>(
+  (
+    columnIndex: number,
+    proposedSortOrder: SortOrder,
+    event: React.KeyboardEvent<HTMLElement> | React.MouseEvent<HTMLElement>
+  ) => {
+    tableItems.splice(
```
Comment (evanlouie) [inline src/client/DeploymentTable.tsx:56]: Can we use `tableItems = tableItems.slice(` instead? In general we should prefer immutable transforms (don't know if `ObservableArray` requires mutable transforms -- if so, i'm fine with leaving as is)

File: src/client/DeploymentTable.tsx [hunk 2]
```diff
@@ -330,7 +385,11 @@ export const renderMergedBy = (
   tableColumn: ITableColumn<IDeploymentField>,
   deployment: IDeploymentField
 ): JSX.Element => {
-  if (deployment.pr && deployment.mergedByName) {
+  if (
+    deployment.pr &&
+    deployment.mergedByName &&
```
Comment (evanlouie) [inline src/client/DeploymentTable.tsx:390]: I'd recommend keeping as `if (deployment.pr && deployment.mergedByName)` as its more concise and does the same thing (empty string is falsy)

**Comment by samiyaakhtar** [comment]
@andrebriggs Deployed a version here http://52.149.54.124:5000/

File: src/client/DeploymentTable.tsx (hunk 1 above)
Comment (edaena) [reply src/client/DeploymentTable.tsx:56]: Observable array does not support slice, only splice.

**Pull request merged**